### DISCUSSION TITLE
Installed several WP sites, all WP sites do not show PHP version or WP version

### DISCUSSION THREAD

**Discussion by beemerrider** [akeeba/panopticon#800] [Q&A (Community Support)] [answered]
All Joomla sites show PHP and Joomla version. Wordpress sites do not and they all throw the same error when you click the site name. Click the troubleshooting button and it says: All good!
The connection to your site was tested, and found to be working fine.

TypeError: Akeeba\Panopticon\Library\SoftwareVersions\WordPressVersion::isEOL(): Argument #1 ($version) must be of type string, null given, called in /home/formations6586/public_html/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php on line 116 and defined in /home/formations6586/public_html/src/Library/SoftwareVersions/WordPressVersion.php:148 Stack trace: #0 /home/formations6586/public_html/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php(116): Akeeba\Panopticon\Library\SoftwareVersions\WordPressVersion->isEOL(NULL) #1 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1301): include('/home/formation...') #2 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1231): Awf\Mvc\View->includeTemplateFile(Array) #3 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(745): Awf\Mvc\View->processTemplate(Array) #4 /home/formations6586/public_html/tmp/compiled_templates/b/f/bfd909ab70a5600336cd8e46959d6dfbf9a45b3c.php(91): Awf\Mvc\View->loadAnyTemplate('Sites/item_wpup...') #5 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1301): include('/home/formation...') #6 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1231): Awf\Mvc\View->includeTemplateFile(Array) #7 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(745): Awf\Mvc\View->processTemplate(Array) #8 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(638): Awf\Mvc\View->loadAnyTemplate('Sites/item') #9 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(562): Awf\Mvc\View->loadTemplate(NULL) #10 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(447): Awf\Mvc\View->display() #11 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(113): Awf\Mvc\Controller->display() #12 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(375): Awf\Mvc\DataController->read() #13 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(57): Awf\Mvc\Controller->execute('read') #14 /home/formations6586/public_html/src/Controller/Sites.php(64): Awf\Mvc\DataController->execute('read') #15 /home/formations6586/public_html/vendor/akeeba/awf/src/Dispatcher/Dispatcher.php(125): Akeeba\Panopticon\Controller\Sites->execute('read') #16 /home/formations6586/public_html/src/Dispatcher.php(36): Awf\Dispatcher\Dispatcher->dispatch() #17 /home/formations6586/public_html/vendor/akeeba/awf/src/Application/Application.php(247): Akeeba\Panopticon\Dispatcher->dispatch() #18 /home/formations6586/public_html/src/Application.php(430): Awf\Application\Application->dispatch() #19 /home/formations6586/public_html/index.php(35): Akeeba\Panopticon\Application->dispatch() #20 /home/formations6586/public_html/index.php(38): {closure}() #21 {main} Next ErrorException: Akeeba\Panopticon\Library\SoftwareVersions\WordPressVersion::isEOL(): Argument #1 ($version) must be of type string, null given, called in /home/formations6586/public_html/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php on line 116 [View template: ROOT/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php line 148, compiled from ROOT/ViewTemplates/Sites/item_wpupdate.blade.php] in /home/formations6586/public_html/src/Library/SoftwareVersions/WordPressVersion.php:148 Stack trace: #0 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1235): Awf\Mvc\View->handleViewException(Object(TypeError), 2) #1 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(745): Awf\Mvc\View->processTemplate(Array) #2 /home/formations6586/public_html/tmp/compiled_templates/b/f/bfd909ab70a5600336cd8e46959d6dfbf9a45b3c.php(91): Awf\Mvc\View->loadAnyTemplate('Sites/item_wpup...') #3 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1301): include('/home/formation...') #4 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1231): Awf\Mvc\View->includeTemplateFile(Array) #5 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(745): Awf\Mvc\View->processTemplate(Array) #6 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(638): Awf\Mvc\View->loadAnyTemplate('Sites/item') #7 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(562): Awf\Mvc\View->loadTemplate(NULL) #8 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(447): Awf\Mvc\View->display() #9 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(113): Awf\Mvc\Controller->display() #10 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(375): Awf\Mvc\DataController->read() #11 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(57): Awf\Mvc\Controller->execute('read') #12 /home/formations6586/public_html/src/Controller/Sites.php(64): Awf\Mvc\DataController->execute('read') #13 /home/formations6586/public_html/vendor/akeeba/awf/src/Dispatcher/Dispatcher.php(125): Akeeba\Panopticon\Controller\Sites->execute('read') #14 /home/formations6586/public_html/src/Dispatcher.php(36): Awf\Dispatcher\Dispatcher->dispatch() #15 /home/formations6586/public_html/vendor/akeeba/awf/src/Application/Application.php(247): Akeeba\Panopticon\Dispatcher->dispatch() #16 /home/formations6586/public_html/src/Application.php(430): Awf\Application\Application->dispatch() #17 /home/formations6586/public_html/index.php(35): Akeeba\Panopticon\Application->dispatch() #18 /home/formations6586/public_html/index.php(38): {closure}() #19 {main} Next ErrorException: Akeeba\Panopticon\Library\SoftwareVersions\WordPressVersion::isEOL(): Argument #1 ($version) must be of type string, null given, called in /home/formations6586/public_html/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php on line 116 [View template: ROOT/tmp/compiled_templates/c/f/cf2d95dc1f05a039c614ac753a57a5b248ce8aeb.php line 148, compiled from ROOT/ViewTemplates/Sites/item_wpupdate.blade.php] in /home/formations6586/public_html/src/Library/SoftwareVersions/WordPressVersion.php:148 Stack trace: #0 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(1235): Awf\Mvc\View->handleViewException(Object(ErrorException), 1) #1 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(745): Awf\Mvc\View->processTemplate(Array) #2 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(638): Awf\Mvc\View->loadAnyTemplate('Sites/item') #3 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/View.php(562): Awf\Mvc\View->loadTemplate(NULL) #4 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(447): Awf\Mvc\View->display() #5 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(113): Awf\Mvc\Controller->display() #6 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/Controller.php(375): Awf\Mvc\DataController->read() #7 /home/formations6586/public_html/vendor/akeeba/awf/src/Mvc/DataController.php(57): Awf\Mvc\Controller->execute('read') #8 /home/formations6586/public_html/src/Controller/Sites.php(64): Awf\Mvc\DataController->execute('read') #9 /home/formations6586/public_html/vendor/akeeba/awf/src/Dispatcher/Dispatcher.php(125): Akeeba\Panopticon\Controller\Sites->execute('read') #10 /home/formations6586/public_html/src/Dispatcher.php(36): Awf\Dispatcher\Dispatcher->dispatch() #11 /home/formations6586/public_html/vendor/akeeba/awf/src/Application/Application.php(247): Akeeba\Panopticon\Dispatcher->dispatch() #12 /home/formations6586/public_html/src/Application.php(430): Awf\Application\Application->dispatch() #13 /home/formations6586/public_html/index.php(35): Akeeba\Panopticon\Application->dispatch() #14 /home/formations6586/public_html/index.php(38): {closure}() #15 {main}

**Comment by nikosdion** [comment]
First, please make sure you are using Panopticon 1.2.2.

In the main page click on the first reload button. See below.
![image](https://github.com/user-attachments/assets/d9000e12-ce98-4164-853a-51a9d16e3f76)

If this does not work, uninstall and reinstall the connector plugin on your site.

**Comment by beemerrider** [comment]
I am running the latest version of Panopticon 1.2.2 and also the latest plugin for Wordpress. I uninstalled the plugin and reinstalled the plugin. No issues installing the plugin on the site. Refreshed the site in the admin of Panopticon.  I have five WP sites all with the same issue. All the 17 Joomla sites that work fine in Panopticon. All sites are on the same dedicated server.

I looked at Mod security logs and did not find anything blocking in there thinking something might be blocking the communication. Although the troubleshooting connector tab in Panopticon says all good.

My Joomla sites are all configured pretty much the same for basic extensions. However the WP sites all vary with the exception of Akeeba Backup Pro and Akeeba Admin that are installed on all of the WP sites. WP is a second language to me, so that does not help. I’ve been using Joomla since 2010, so I’m much better with that CMS.

Anything you can think of in my server config that may be interfering with these WP sites. I have full access to the server to change anything required.



Bruce Wilson

360-931-3127

**Reply by nikosdion** [reply]
I can help you troubleshoot your issue, but I need the following connection information to your Panopticon site.

**Site connection information**

* Login URL e.g. `https://www.example.com`
* Username
* Password

**(S)FTP connection information**

* Hostname
* Port (if other than 21 or 22)
* Username
* Password

Do **NOT** paste this information unencrypted. GitHub issues and discussions are publicly visible. Instead, follow the instructions below to send me an encrypted copy of the information which can only be decrypted by me:

* Go to https://keybase.io/encrypt
* In the **Recipient** field type `nikosdion`.
* It will show you my face, my name, and the fact that I have linked my same-named GitHub account; verify this information and click on my face.
* In the **Message to encrypt** area enter the information I asked you above.
* Click the Encrypt button.
* After a short while the text area changes its title to **The secret message**.
* Click in the text area to copy the secret (encrypted) message.
* Paste it to your next reply to me.

This is perfectly safe to do! It uses public key encryption, just like HTTPS, with a _much, MUCH stronger cipher_ (3744-bit Elliptic Curve Cryptography which is insanely secure). The message can only be decrypted with my secret key which is safely stored on a hardware token, protected by a PIN.

Please make sure that you test the access credentials using a separate browser / (S)FTP software than what you normally use to ensure that they really work. Please make sure that there are no geographic or other restrictions which may prevent me from accessing your site. Kindly note that I am on the [Asia/Nicosia timezone](https://www.timeanddate.com/worldclock/cyprus/nicosia) and I am handling support free of charge; some delays are to be expected.

**Comment by beemerrider** [comment]
Thanks for the excellent support. I hope you have a paid version some day so I can pay you.


Bruce Wilson

1+360-931-3127

-----BEGIN PGP MESSAGE-----
Version: Keybase OpenPGP v2.1.13
Comment: https://keybase.io/crypto

wcBMA6+1LLjxrhABAQf/UhUl/GVTR08Ch2DBQ+mkVCgdffJJLolw7c3miBmMTGqc
C9tguZKSwpmZcoI1yv9Ju/hLulGaFYx5hr7QktK+IyRPTcGQk8XmpwZw7js7ujq3
dtAkmzWn0+yAfbcWji57gf8Q8dK3vUo5nEk8zV94wWUcViT7XfJKnxyMc3fxQurE
+7YjZdecaBr1z97ZwUSOcabLFmeqyxnbfG+w9VhNPasqEQl0oUv6ZAAZTqAZbGT5
AgJS76FXeZxHkd48BkA4B7cu4mnpaMr18NGvJ4TPBCIrxaTQCYSKUSsg2sWyUNlJ
Edk+2oTKltt3W+2hYq6rAHvHgqTNXcXTtOqztpEKEtLAIwFRec/cc+7gSnQsArDE
aXqXvXruMJu7jKZSVK16kYcWJrbgh59jwNRIDV9+Ki+wsrDzST9Cysr2jsl5lU+c
YwFMF6hYWUdyK47cOFkuPIQr3R/b1RR5XAyz+Qb5eGfn03tj+6COSxpw2tBDBlKl
lXnU8fyv90cVlojnhciCT4z/be2U8SHDVDINVl58CPORwjMLcSzltFshNZXsnXr4
AT351BRIsN/X+dm7ZSzOzt6F+NPd1QoHCeD/xGTHff71Us1hao1BApiQ0sKEAgMk
n4gasJWJKssGlqIPefJgz2pmQyf8
=kSTc
-----END PGP MESSAGE-----

**Reply by nikosdion** [reply]
The problem I see is that the WordPress site cannot load the connector plugin. Try reinstalling the plugin and make sure that all files have been copied. The file it complains is missing is part of the ZIP file we released.

**Reply by beemerrider** [reply, marked as answer]
I had reinstalled the plugin previously and that had not fixed the issue. So this time I downloaded a fresh copy of the plugin and that solved the problem of fully connecting. I went back and looked at the two versions I downloaded and they were completely different. I'm not sure where I went wrong the first time. The names of the plugins were also different. Bad one "panopticon-connector-wordpress-main.zip" and the correct one, "panopticon-1.0.1.zip:

Thanks for the help. I noticed in the readme there was mention of a Paypal donating link. Can you reply with the donation link please?

**Reply by nikosdion** [reply]
Ah, you got trolled by the GitHub releases interface. 

The `panopticon-connector-wordpress-main.zip` file is the source code. This does not have the `vendor` folder; you'd have to run `composer install` in its directory for Composer to pull the dependencies and install them in the `vendor` folder.

The `panopticon-1.0.1.zip` file is the _actual_ plugin installable package. This is generated from my working copy after I've already run `composer install`. It also doesn't have files you don't need, like the build script.

Unfortunately, GitHub Releases doesn't let me hide the former, or even label it in a different way. WordPress is also dumb enough that it will accept that file which is very clearly NOT a WordPress plugin installation ZIP file without complaining. Sigh.
